practice/flutter/flutter/views.py

```
@@ -0,0 +1,219 @@
+    rq_data.update(get_request_data(request.GET, {'next': 'home'})[0])
+
+    if request.user.is_authenticated:
+        success = True
+        rq_data['username'] = request.user.username
+    elif is_post:
+        if auth_user(request, rq_data['username'], rq_data['password']):
+            success = True
+        else:
+            err_id = 'bad_login'
+
+    if success:
+        return redirect(rq_data['next'])
+    else:
+        view_data = logic.get_error_message_and_status(err_id)
+        view_data.update(rq_data)
+        return render_template_and_data(request, 'login', view_data)
+
+
+def logout(request: HttpRequest) -> render:
+    """ Destroy user session """
+    deauth_user(request)
+    return render_template_and_data(request, 'logout', {})
+
+
+def user_profile(request: HttpRequest, user_name: str) -> render:
+    """ Display a specific user """
+    found = True
+    template = 'profile'
+
+    try:
+        view_data = {'profile': logic.get_user_by_username(user_name)}
+    except LookupError:
+        return render_not_found(request, 'user', user_name, {})
+
+    if found:
+        view_data['member_since'] = arrow.get(
+            view_data['profile'].date_joined
+        ).humanize()
+
+        try:
+            view_data['flutts'] = formatter.format_flutts(
+                logic.get_flutts_by_user(user=view_data['profile'])
+            )
+        except KeyError:
+            view_data['flutts'] = []
+
+    return render_template_and_data(request, template, view_data)
+
+
+# I already kind of knew about decorators.
+# Still trying to wrap my head around annotations
+# https://github.com/champs794/codeguild/pull/20#discussion_r78058649
+#
+@login_required
+def user_home(request: HttpRequest) -> render:
+    """ Display user home page """
+    err_id = False
+    text = request.POST.get('text')
+
+    if text is None:
+        flutt = text
+    else:
+        try:
+            flutt = logic.save_flutt(text, request.user)
```

The stuff about silently passing None.